azure/rs_azure_compute/azure_compute_plugin.rb

```diff
@@ -0,0 +1,208 @@
+name 'rs_azure_compute'
+type 'plugin'
+rs_ca_ver 20161221
+short_description "Azure Compute Plugin"
+long_description "Version: 1.0"
+package "plugins/rs_azure_compute"
+import "sys_log"
+
+parameter "subscription_id" do
+  type  "string"
+  label "Subscription ID"
+end
+
+permission "read_creds" do
+  actions   "rs_cm.show_sensitive","rs_cm.index_sensitive"
+  resources "rs_cm.credentials"
+end
+
+plugin "rs_azure_compute" do
+  endpoint do
+    default_host "https://management.azure.com/"
+    default_scheme "https"
+    query do {
+      'api-version' =>  '2016-04-30-preview'
+    } end
+  end
+
+  parameter "subscription_id" do
+    type  "string"
+    label "subscription_id"
+  end
+
+  type "availability_set" do
+    href_templates "{{id}}"
+    provision "provision_resource"
+    delete    "delete_resource"
+
+    field "properties" do
+      type "composite"
+      location "body"
+    end
+
+    field "location" do
+      type "string"
+      location "body"
+    end
+
+    field "resource_group" do
+      type "string"
+      location "path"
+    end 
+
+    field "name" do
+      type "string"
+      location "path"
+    end
+    
+    field "tags" do
+      type "composite"
+      location "body"
+    end
+
+    field "sku" do
+      type "composite"
+      location "body"
+    end
+
+    action "create" do
+      type "availability_set"
+      path "/subscriptions/$subscription_id/resourceGroups/$resource_group/providers/Microsoft.Compute/availabilitySets/$name"
+      verb "PUT"
+    end
+
+    action "show" do
+      type "availability_set"
+      path "$href"
+      verb "GET"
+    end
+
+    action "get" do
+      type "availability_set"
+      path "$href"
+      verb "GET"
+    end
+
+    action "destroy" do
+      type "availability_set"
+      path "$href"
+      verb "DELETE"
+    end
+    
+    output "virtualmachines" do
+      body_path "properties.virtualMachines"
+    end
+
+    output "id","name","location","tags","sku","properties"
+  end
+
+  type "virtualmachine" do
+    href_templates "{{contains(id, 'virtualMachines') && id || null}}"
+    provision "no_operation"
+    delete    "no_operation"
+
+    field "properties" do
+      type "composite"
+      location "body"
+    end
+
+    field "location" do
+      type "string"
+      location "body"
+    end
+
+    field "resource_group" do
+      type "string"
+      location "path"
+    end
+
+    action "show" do
```

Comment:
let's add "resource_group" and "vm_name" as allowed fields to this call & update the path.  thus allowing code like this: `rs_azure_compute.virtualmachine.show(resource_group: "foo", vm_name: "bar")`